src/context/context.tsx

```diff
@@ -0,0 +1,20 @@
+import React, { createContext, useContext, useReducer } from "react";
+import { authReducer, initialAuthState } from "../reducer/authReducer";
+
+const AuthContext = createContext({} as ContextValue);
```

Comment:
I guess we can initialize context with initialState and a dispatch function that returns null. This type assertion seems little wrong.

Reply:
Will look into it, and use the better approach.